prolog/st/st_tokens.pl

```diff
@@ -21,11 +21,35 @@
 
 st_tokens(Codes, Options, Tokens):-
     option(frontend(Frontend), Options),
-    must_be(oneof([simple, semblance]), Frontend),
-    phrase(tokens(Tmp1, Frontend), Codes),
+    (( Frontend =
+            syntax_tokens(
+                comment(_, _),
+                out(_, _),
+                out_unescaped(_, _),
+                statement(_, _)),
+        SyntaxTokens = Frontend
+    ) ; ( 
+        must_be(oneof([simple, semblance]), Frontend),
+        named_syntax_tokens(Frontend, SyntaxTokens)
+    )),
+    phrase(tokens(Tmp1, SyntaxTokens), Codes),
     phrase(collapse(Tmp2), Tmp1), !,
     Tokens = Tmp2.
 
+named_syntax_tokens(simple, syntax_tokens(
+    comment("{{%","}}"),
+    out("{{=","}}"),
+    out_unescaped("{{-","}}"),
+    statement("{{","}}"))).
+
+named_syntax_tokens(semblance, syntax_tokens(
+    comment("{#","#}"),
+    out("{{","}}"),
+    out_unescaped(keyword_unescape_start("{%"),"%}"),
+    statement("{%","%}"))).
+
+keyword_unescape_start(StartToken) --> StartToken, whites, "unescape".
```

Comment:
Moved the `simple` and `semblance` definitions here into terse data used by the DCG rules.

Reply:
This removes many deduplicates other frontend specific rules below, and passes the existing test suite.